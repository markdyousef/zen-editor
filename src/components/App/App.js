import React, { Component } from 'react';
import { Editor, EditorState, RichUtils } from 'draft-js';
import 'draft-js/dist/Draft.css';
import styled from 'styled-components';
import Toolbar from '../Toolbar';
import FloatingActionButton from '../FloatingActionButton';
import customRenderer from '../../utils/customRenderer';

const Container = styled.div`
    padding: 30px 30px;
    box-sizing: border-box;
    position: relative;
`;

const EditorContainer = styled.div`
    cursor: text;
    position: relative;
    margin: 0 auto;
    margin-top: 10px;
    max-width: 700px;
`;

export default class App extends Component {
    constructor() {
        super();
        this.state = {
            editorState: EditorState.createEmpty()
        };
    }
    onChange = editorState => this.setState({ editorState })
    focus = () => this.editor.focus();
    toggleBlockType = (blockType) => {
        const { editorState } = this.state;
        this.onChange(
            RichUtils.toggleBlockType(
                editorState,
                blockType
            )
        );
    }
    toggleInlineStyle = (inlineStyle) => {
        const { editorState } = this.state;
        this.onChange(
            RichUtils.toggleInlineStyle(
                editorState,
                inlineStyle
            )
        );
    }
    render() {
        const { editorState } = this.state;
        const showToolbar = !editorState.getSelection().isCollapsed();
        return (
            <Container>
                <EditorContainer>
                    <Editor
                        ref={(node) => { this.editor = node; }}
                        editorState={editorState}
                        spellCheck
                        placeholder="Write something cool..."
                        onChange={this.onChange}
                        blockRendererFn={customRenderer(editorState, this.onChange)}
                    />
                    <FloatingActionButton
                        editorState={editorState}
                        focus={this.focus}
                        setEditorState={this.onChange}
                    />
                    <Toolbar
                        editorState={editorState}
                        toggleBlockType={this.toggleBlockType}
                        toggleInlineStyle={this.toggleInlineStyle}
                        focus={this.focus}
                        editorNode={this.editor}
                        showToolbar={showToolbar}
                    />
                </EditorContainer>
            </Container>
        );
    }
}
