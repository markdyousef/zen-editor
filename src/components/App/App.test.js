import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders component correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);

    const component = renderer.create(<App />);
    expect(component).toMatchSnapshot(component);
});
