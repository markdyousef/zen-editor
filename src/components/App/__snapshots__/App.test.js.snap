exports[`[object Object] 1`] = `
<div
  className="lcrruv">
  <div
    className="hSmQAk">
    <div
      className="DraftEditor-root">
      <div
        className="public-DraftEditorPlaceholder-root">
        <div
          className="public-DraftEditorPlaceholder-inner"
          id="placeholder-5bkb3">
          Write something cool...
        </div>
      </div>
      <div
        className="DraftEditor-editorContainer">
        <div
          aria-activedescendant={undefined}
          aria-autocomplete={undefined}
          aria-describedby="placeholder-5bkb3"
          aria-expanded={undefined}
          aria-haspopup={undefined}
          aria-label={undefined}
          aria-owns={undefined}
          className="public-DraftEditor-content"
          contentEditable={true}
          data-testid={undefined}
          onBeforeInput={[Function]}
          onBlur={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onCopy={[Function]}
          onCut={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseUp={[Function]}
          onPaste={[Function]}
          onSelect={[Function]}
          role="textbox"
          spellCheck={true}
          style={
            Object {
              "outline": "none",
              "whiteSpace": "pre-wrap",
              "wordWrap": "break-word",
            }
          }
          suppressContentEditableWarning={true}
          tabIndex={undefined}>
          <div
            data-contents="true">
            <div
              className=""
              data-block={true}
              data-editor="5bkb3"
              data-offset-key="dg0r1-0-0">
              <div
                className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                data-offset-key="dg0r1-0-0">
                <span
                  data-offset-key="dg0r1-0-0"
                  style={Object {}}>
                  <br
                    data-text="true" />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="gACtEH">
      <div>
        <span
          className="kLKimt"
          onClick={[Function]}>
          H1
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          H2
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          H3
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          Quote
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          Bulleted List
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          Numbered List
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          Code Block
        </span>
      </div>
      <div>
        <span
          className="kLKimt"
          onClick={[Function]}>
          B
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          I
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          U
        </span>
        <span
          className="kLKimt"
          onClick={[Function]}>
          Hi
        </span>
      </div>
    </div>
  </div>
</div>
`;
